### PR TITLE
Fix issue around checking whether the Activity type is available in ActivityExtensions.

### PR DESCRIPTION
**Pull request by macrogreg**
Fix issue around checking whether the Acitity type is available in ActivityExtensions (https://github.com/microsoft/ApplicationInsights-dotnet/issues/1761).

### CODE REVIEW

**Comment by cijothomas** [comment]
Thanks @macrogreg . I'll wait to hear from @SergeyKanzhelev if this TryRun approach can be removed entirely.

I believe the whole idea of TryRun is to enable scenario where AI.dll can run standalone without System.DiagnosticsSource.dll. I wonder if this is even a valid scenario? For anyone installing application insights via nuget, System.DiagnosticsSource will be installed by nuget. 
The only scenario I can think of is - if someone manually copies the AI.DLL file for use from a powershell module?

@SergeyKanzhelev Do you think if we should still support this? Every time we upgrade System.DiagnosticSource package version, we keep updating this file.

File: BASE/src/Microsoft.ApplicationInsights/ActivityExtensions.cs [hunk 1]
```diff
@@ -47,29 +50,62 @@ internal static void SetOperationName(this Activity activity, string operationNa
...
+                return false;
             }
         }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static bool LoadActivityType()
+        {
+            // At this point the Activity type has already been loaded, or - if not - the runtime will attempt to load the assembly from which it is referenced.
+            // While doing so, it will respect type redirects potentially set up by the execution environment. This is preferable to doing Assembly-Load directly.
+            return typeof(System.Diagnostics.Activity) != null;
```
Comment (cijothomas) [inline BASE/src/Microsoft.ApplicationInsights/ActivityExtensions.cs:108]: SDK must have a particular version of the assembly, as it relies of features not available in older versions. (like w3c format etc.) This was probably the motivation behind original code trying to load the exact same version.
With this code, we may break else where, if an Activity of older version is available, and we attempt to call methods not available in that version.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ## VNext
 
 ## Version 2.14.0-beta4
+- [Fix: https://github.com/microsoft/ApplicationInsights-dotnet/issues/1761](https://github.com/microsoft/ApplicationInsights-dotnet/issues/1761)
```
Comment (TimothyMothra) [inline CHANGELOG.md:6]: Please move this to the VNext section. Beta4 has already been compiled and will be released soon.

**Comment by cijothomas** [comment]
Closing this PR. Had synced with Sergey and decided that we wont need to support the standalone usage of ai.dll. That makes this PR irrelevant, as SDK would no longer attempt to load DiagnosticSource.

**Pull request closed without merging**